lib/node_modules/@stdlib/blas/ext/index-of/lib/main.js

```diff
@@ -0,0 +1,204 @@
+
+// MODULES //
+
+var hasOwnProp = require( '@stdlib/assert/has-own-property' );
+var isNumber = require( '@stdlib/assert/is-number' ).isPrimitive;
+var isndarrayLike = require( '@stdlib/assert/is-ndarray-like' );
+var broadcastScalar = require( '@stdlib/ndarray/base/broadcast-scalar' );
+var maybeBroadcastArray = require( '@stdlib/ndarray/base/maybe-broadcast-array' );
+var getDType = require( '@stdlib/ndarray/dtype' );
+var getShape = require( '@stdlib/ndarray/shape' );
+var getOrder = require( '@stdlib/ndarray/order' );
+var format = require( '@stdlib/string/format' );
+var defaults = require( '@stdlib/ndarray/defaults' );
+var nonCoreShape = require( './non_core_shape.js' );
+var base = require( './base.js' );
+
+
+// VARIABLES //
+
+var DEFAULT_DTYPE = defaults.get( 'dtypes.integer_index' );
+
+
+// MAIN //
+
+/**
+* Returns the first index of a specified search element along one or more ndarray dimensions.
+*
+* @name indexOf
+* @type {Function}
+* @param {ndarrayLike} x - input ndarray
+* @param {(ndarrayLike|number)} searchElement - search element
+* @param {(ndarrayLike|number)} [fromIndex] - index from which to begin searching
+* @param {Options} [options] - function options
+* @param {IntegerArray} [options.dims] - list of dimensions over which to perform operation
+* @param {boolean} [options.keepdims=false] - boolean indicating whether the reduced dimensions should be included in the returned ndarray as singleton dimensions
+* @param {string} [options.dtype] - output ndarray data type
+* @throws {TypeError} first argument must be an ndarray-like object
+* @throws {TypeError} search element argument must be either an ndarray-like object or a numeric value
+* @throws {TypeError} from index argument must be either an ndarray-like object or a numeric value
+* @throws {TypeError} options argument must be an object
+* @throws {RangeError} dimension indices must not exceed input ndarray bounds
+* @throws {RangeError} number of dimension indices must not exceed the number of input ndarray dimensions
+* @throws {Error} must provide valid options
+* @returns {ndarray} output ndarray
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+* var ndarray = require( '@stdlib/ndarray/ctor' );
+*
+* // Create a data buffer:
+* var xbuf = new Float64Array( [ 1.0, 2.0, -3.0, 4.0, -5.0, 6.0 ] );
+*
+* // Define the shape of the input array:
+* var sh = [ 3, 1, 2 ];
+*
+* // Define the array strides:
+* var sx = [ 2, 2, 1 ];
+*
+* // Define the index offset:
+* var ox = 0;
+*
+* // Create an input ndarray:
+* var x = new ndarray( 'float64', xbuf, sh, sx, ox, 'row-major' );
+*
+* // Find index:
+* var out = indexOf( x, 4.0 );
+* // returns <ndarray>
+*
+* var idx = out.get();
+* // returns 3
+*/
+function indexOf( x, searchElement, fromIndex, options ) {
+	var nargs;
+	var opts;
+	var ord;
+	var dt;
+	var sh;
+
+	nargs = arguments.length;
+
+	// Resolve input ndarray meta data:
+	dt = getDType( x );
+	ord = getOrder( x );
+
```

Comment:
Explicitly handle the case where `nargs < 2` first. Right now, this gets handled with `nargs < 4` below which just leads to incorrect error messages.